plugins/woocommerce-blocks/tests/e2e/tests/rating-filter/rating-filter.block_theme.spec.ts

```diff
@@ -0,0 +1,280 @@
+
+		await expect( page.url() ).toContain(
+			blockData.urlSearchParamWhenFilterIsApplied
+		);
+
+		await expect( products ).toHaveCount( 1 );
+	} );
+} );
+
+test.describe( `${ blockData.name } Block - with Product Collection`, () => {
+	test.beforeEach(
+		async ( {
+			admin,
+			editorUtils,
+			productCollectionPageObject,
+			editor,
+		} ) => {
+			await admin.createNewPost();
+			await productCollectionPageObject.insertProductCollection();
+			await productCollectionPageObject.chooseCollectionInPost(
+				'productCatalog'
+			);
+			await editor.insertBlock( {
+				name: 'woocommerce/filter-wrapper',
+				attributes: {
+					filterType: 'rating-filter',
+					heading: 'Filter By Rating',
+				},
+			} );
+			await editorUtils.publishAndVisitPost();
+		}
+	);
+
+	test( 'should show all products', async ( { page } ) => {
+		const products = page
+			.locator( '.wp-block-woocommerce-product-template' )
+			.getByRole( 'listitem' );
+
+		await expect( products ).toHaveCount( 9 );
+	} );
+
+	test( 'should show only products that match the filter', async ( {
+		page,
+	} ) => {
+		await page
+			.getByRole( 'checkbox', { name: 'Rated 1 out of 5' } )
+			.click();
+
+		await page.waitForURL( ( url ) =>
+			url
+				.toString()
+				.includes( blockData.urlSearchParamWhenFilterIsApplied )
+		);
+
+		const products = page
+			.locator( '.wp-block-woocommerce-product-template' )
+			.getByRole( 'listitem' );
+
+		await expect( products ).toHaveCount( 1 );
+	} );
+
+	test( 'should refresh the page only if the user click on button', async ( {
+		page,
+		admin,
+		editor,
+		editorUtils,
+		productCollectionPageObject,
+	} ) => {
+		await admin.createNewPost();
+		await productCollectionPageObject.insertProductCollection();
+		await productCollectionPageObject.chooseCollectionInPost(
+			'productCatalog'
+		);
+		await editor.insertBlock( {
+			name: 'woocommerce/filter-wrapper',
+			attributes: {
+				filterType: 'rating-filter',
+				heading: 'Filter By Rating',
+			},
+		} );
+
+		const ratingFilterControls = await editorUtils.getBlockByName(
+			'woocommerce/rating-filter'
+		);
+		await editor.selectBlocks( ratingFilterControls );
+		await editor.openDocumentSettingsSidebar();
+		await page.getByText( "Show 'Apply filters' button" ).click();
+		await editorUtils.publishAndVisitPost();
+
+		await page.addInitScript( () => {
+			document.addEventListener( 'DOMContentLoaded', () => {
+				// eslint-disable-next-line dot-notation
+				window[ '__DOMContentLoaded__' ] = true;
+			} );
+		} );
+
+		await page
+			.getByRole( 'checkbox', { name: 'Rated 1 out of 5' } )
+			.click();
+		await page.getByRole( 'button', { name: 'Apply' } ).click();
+
+		await page.waitForEvent( 'domcontentloaded' );
+
+		const domContentLoaded = await page.evaluate(
+			// eslint-disable-next-line dot-notation
+			() => window[ '__DOMContentLoaded__' ] === true
+		);
+
+		await expect( page.url() ).toContain(
+			blockData.urlSearchParamWhenFilterIsApplied
+		);
```

Comment:
Let's use the dedicated assertion here:

```suggestion
		await expect( page ).toHaveURL(
			new RegExp( blockData.urlSearchParamWhenFilterIsApplied )
		);
```